### DISCUSSION TITLE
Extension crashes and won't run

### DISCUSSION THREAD

**Discussion by Augs0** [vitest-dev/vscode#450] [Q&A] [answered]
Hello! I am having some trouble running the extension.

VS Code version:  1.92.1
Vitest version: 1.4.0
Node version: 18.19.0
OS: MacOS Sonoma 14.5

I seem to have the requirements needed, but as soon as I open my repo, I get an error message telling me something went wrong.
![Screenshot 2024-08-13 at 09 19 50](https://github.com/user-attachments/assets/9bd1c7bc-a9d1-4e67-9c27-a7ad17f21a1f)

When I click to see the error, I get a bunch of failures in the `extension.js` file, specifically around a `string.replace`
![Screenshot 2024-08-13 at 09 20 11](https://github.com/user-attachments/assets/7a519125-5eba-4a87-934f-8ed449fad740)

Has anyone seen this before? Can you recommend how to fix this? I have tried updating Vitest, but the same thing occurs and 1.4.0 is the minimum at which this should work, I believe.

**Comment by pelargir** [comment]
I'm getting the exact same error. Baffled as to how to fix it. Any help?

**Reply by pelargir** [reply, marked as answer]
I finally figured out why this was happening.

I had recently renamed my all-lowercase project directory to use camel-case instead (for example, "myproject" became "myProject"). 

VSCode still had the path as "myproject" so the code in the vitest extension that's designed to detect and avoid recursion when scanning for test files was unable to tell the difference and kept recursing down until it got to the root of my HD.

Removing the project from VSCode and creating a new project by selecting the renamed directory fixed the problem.

**Reply by Augs0** [reply]
This worked for me too! Thank you for coming back and sharing!
